tensorboard/backend/event_processing/io_wrapper.py

```diff
@@ -162,6 +167,59 @@ def ListRecursivelyViaGlobbing(top):
         current_glob_string = os.path.join(current_glob_string, "*")
         level += 1
 
+import queue
+import time
+import os
+import threading
+class listdir_bfs:
+    def __init__(self, baseDir, list_dir_function=None):
```

Comment:
can you add docstring on param types and explanation? It'll also be helpful to explain data structure for self. return_structure